compiler/src/dotty/tools/dotc/core/GadtConstraint.scala

```diff
@@ -71,12 +71,33 @@ class GadtConstraint private (
     externalize(constraint.nonParamBounds(param)).bounds
 
   def fullLowerBound(param: TypeParamRef)(using Context): Type =
-    constraint.minLower(param).foldLeft(nonParamBounds(param).lo) {
+    val self = externalize(param)
+    constraint.minLower(param).filterNot { p =>
```

Comment:
Nitpick: I find the double-negation  `filterNot { ... !hi.isExactlyAny ... }` a bit hard to read, I'll let you decide but I think I would prefer a `filter`. Maybe we could also do the filtering inline inside the foldLeft lambda to avoid an extra-traversal if it doesn't hurt readability.